File: Main-1.java
/*
Akulkumar Patel
CS 280, 450
NJIT-ID: 31504835
*/

import java.util.*;

public class Main
{
    private static final int N = 8;
    private int tour[][];
    
    public Main()
    {
        tour = new int[N][N];
    }

    private boolean movePossible(int x, int y)
    {
        if (x >= 0 && x < N && y >= 0 && y < N && tour[x][y] == 0)
            return true;
        return false;
    }

    private void printKT()
    {
        for (int x = 0; x < N; x++)
        {
            for (int y = 0; y < N; y++)
            {
                System.out.printf("%3d", tour[x][y] +1);
            }
            System.out.println();
        }
    }   

    private boolean findTour(int x, int y, int move_count, int xMove[],int yMove[])
    {
        int i, next_x, next_y;
        if (move_count == N * N)
        return true;

        for (i = 0; i < N; i++)
        {
            next_x = x + xMove[i];
            next_y = y + yMove[i];
            if (movePossible(next_x, next_y))
            {
                tour[next_x][next_y] = move_count;
                if (findTour(next_x, next_y, move_count + 1, xMove, yMove))
                    return true;
                else
                    tour[next_x][next_y] = 0;
            }
        }
        return false;
    }

    public boolean knightTours()
    {
        for (int x = 0; x < N; x++)
        {
            for (int y = 0; y < N; y++)
            {
                tour[x][y] = 0;
            }
        }
        int xMove[] = { 2, 1, -1, -2, -2, -1, 1, 2 };
        int yMove[] = { 1, 2, 2, 1, -1, -2, -2, -1 };
        tour[0][0] = 0;
        if (!findTour(0, 0, 1, xMove, yMove))
        {
            System.out.println("Knights tour does not exist");
            return false;
        }
        else
        {
            printKT();
            return true;
        }
        
    }

    public static void main(String[] arg)
    {
        Main kT = new Main();
        System.out.println("Knights Tour exists...");
        kT.knightTours();
    }
}

File: main-1.cpp
//Akulkumar Patel
//CS 280, 450
//NJIT-ID: 31504835

#include<iostream>
#include<iomanip>
using namespace std;
const int N = 8;

typedef struct chess_moves {
    int x,y;
    }chess_moves;

void printKT(int tour[N][N]) {
   int i,j;
   for (i = 0; i < N; i++) {
      for (j = 0; j < N; j++) {
          cout<<setw(3)<<tour[i][j]+1;
      }
      cout<<endl;
   }
}

bool movePossible(chess_moves next_move, int tour[N][N]) {
   int i = next_move.x;
   int j = next_move.y;
   if ((i >= 0 && i < N) && (j >= 0 && j < N) && (tour[i][j] == 0))
      return true;
   return false;
}

bool findTour(int tour[N][N], chess_moves move_KT[],
               chess_moves curr_move, int move_count)
    {
        int i;
        chess_moves next_move;
        if (move_count == N*N-1)
            {
                return true;
            }

       for (i = 0; i < N; i++)
        {
          next_move.x = curr_move.x + move_KT[i].x;
          next_move.y = curr_move.y + move_KT[i].y;
          if (movePossible(next_move, tour))
          {
             tour[next_move.x][next_move.y] = move_count+1;
             if (findTour(tour, move_KT, next_move, move_count+1) == true)
                return true;
             else
                tour[next_move.x][next_move.y] = 0;
          }
        }
        return false;
    }

void knightTours()
{
   int tour[N][N];
   int i,j;
   for (i = 0; i < N; i++)
    {
      for (j = 0; j < N; j++)
      {
         tour[i][j] = 0;
      }
    }

   chess_moves move_KT[8] = { {2,1},{1,2},{-1,2},{-2,1},
                                {-2,-1},{-1,-2},{1,-2},{2,-1} };

   chess_moves curr_move = {0,0};

   if(findTour(tour, move_KT, curr_move, 0) == false)
    {
      cout<<"Knights tour does not exist";
    }
   else
   {
      cout<<"Knights Tour exists...\n";
      printKT(tour);
   }
}

int main()
{
   knightTours();
   cout<<endl;
   return 0;
}
